examples/data-objects/pond/src/providers/userInfo.ts

```diff
@@ -62,16 +60,11 @@ export class UserInfo extends EventEmitter implements IFluidUserInformation {
     }
 }
 
-export const userInfoFactory = async (dc: DependencyContainer) => {
-    const s = dc.synthesize<IContainerRuntime>({
-        IContainerRuntime,
-        IFluidHandleContext,
-        IFluidDataStoreRegistry,
-    }, {});
+export const userInfoFactory = async (dc: IFluidDependencySynthesizer) => {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    const s = dc.synthesize<{},IProvideContainerRuntime>(
+        {},
+        {IContainerRuntime});
     const containerRuntime = await s.IContainerRuntime;
```

Comment:
Can this be `undefined`?  If so looks like it will error out in constructor, probably prefer to have a human-friendly error here prior to attempting to call the constructor.

Reply:
the typing around the provider model is synthesize is pretty broken. they typing doesn't allow returning undefined, this was sneaking through due to the typing on dependency container entry being incorrect.